lib/HTFeed/Volume.pm

```diff
@@ -28,748 +28,825 @@ my $stage_map = undef;
-    my $class = $self->{nspkg}->get('volume_module');
-
-    bless( $self, $class );
-    return $self;
-  } else {
-    get_logger()->error("BadValue",namespace=>$self->{namespace},objid=>$self->{objid},field=>'barcode',detail=>'Invalid barcode');
-    croak("VOLUME_ERROR");
-  }
+    my $package = shift;
+
+    my $self = {
+	objid       => undef,
+	namespace   => undef,
+	packagetype => undef,
+	uuidgen     => new Data::UUID,
+	@_,
+	# files	    => [],
+	# dir	    => undef,
+	# mets_name => undef,
+	# mets_xml  => undef,
+    };
+
+    $self->{nspkg} = HTFeed::Namespace->new(
+	$self->{namespace},
+	$self->{packagetype}
+    );
+
+    if ($self->{nspkg}->validate_barcode($self->{objid})) {
+	my $class = $self->{nspkg}->get('volume_module');
+	bless($self, $class);
+	return $self;
+    } else {
+	get_logger()->error(
+	    "BadValue",
+	    namespace => $self->{namespace},
+	    objid     => $self->{objid},
+	    field     => 'barcode',
+	    detail    => 'Invalid barcode'
+	);
+	croak("VOLUME_ERROR");
+    }
 }
 
 # invalidate cached information between jobs
 sub reset {
-  my $self = shift;
-  my @fields = qw(checksums content_files source_mets_file source_mets_xpc
-  repsitory_mets_xpc jhove_files utf8_files repository_symlink page_data
-  filegroups ingest_date); 
+    my $self = shift;
 
-  map { delete $self->{$_} } @fields;
+    my @fields = (
+	'checksums',
+	'content_files',
+	'filegroups',
+	'ingest_date',
+	'jhove_files',
+	'page_data',
+	'repository_symlink',
+	'repsitory_mets_xpc',
+	'source_mets_file',
+	'source_mets_xpc',
+	'utf8_files',
+    );
+    map { delete $self->{$_} } @fields;
 }
 
 sub get_identifier {
-  my $self = shift;
-  return $self->get_namespace() . q{.} . $self->get_objid();
-
+    my $self = shift;
+    join(".", $self->get_namespace(), $self->get_objid());
 }
 
 sub get_namespace {
-  my $self = shift;
-  return $self->{namespace};
+    my $self = shift;
+    return $self->{namespace};
 }
 
 sub get_objid {
-  my $self = shift;
-  return $self->{objid};
+    my $self = shift;
+    return $self->{objid};
 }
 
 sub get_packagetype {
-  my $self = shift;
-  return $self->{packagetype};
+    my $self = shift;
+    return $self->{packagetype};
 }
 
 sub get_pt_objid {
-  my $self = shift;
-  return s2ppchars($self->{objid});
+    my $self = shift;
+    return s2ppchars($self->{objid});
 }
 
 sub get_file_groups {
-  my $self = shift;
-
-  if(not defined $self->{filegroups}) {
-    my $filegroups = {}; 
-
-    my $nspkg_filegroups = $self->{nspkg}->get('filegroups');
-    while( my ($key,$val) = each (%{ $nspkg_filegroups })) {
-      my $files = [];
-      my $re = $val->{file_pattern} or die("Missing pattern for filegroup $key");
-      foreach my $file ( @{ $self->get_all_directory_files() }) {
-        push(@$files,$file) if $file =~ $re;
-      }
-      $filegroups->{$key} = new HTFeed::FileGroup($files,%$val);
+    my $self = shift;
+
+    if(not defined $self->{filegroups}) {
```

Comment:
An exception to the "space after if" rule that seems to be in place elsewhere. Looks like there are some other examples in this file too, also I'm in favor of "space after while" and I see an instance of a `while( ...` just below. I think it would be worth doing a search on "if(" and "while(" and aligning the whitespace with the overall standard.